monkey/monkey_island/cc/services/agent_binary_service/flask_resources/agent_binary_masque.py

```diff
@@ -0,0 +1,36 @@
+import logging
+from http import HTTPStatus
+from json import JSONDecodeError
+
+from flask import request
+from flask_security import auth_token_required, roles_accepted
+
+from monkey_island.cc.flask_utils import AbstractResource
+from monkey_island.cc.services.authentication_service import AccountRole
+
+from .. import IAgentBinaryService
+
+logger = logging.getLogger(__name__)
+
+
+class AgentBinaryMasque(AbstractResource):
+    urls = ["/api/agent-binary-masque"]
+
+    def __init__(self, agent_binary_service: IAgentBinaryService):
+        self._agent_binary_service = agent_binary_service
+
+    @auth_token_required
+    @roles_accepted(AccountRole.AGENT.name, AccountRole.ISLAND_INTERFACE.name)
+    def get(self):
+        return self._agent_binary_service.get_masque(), HTTPStatus.OK
+
+    @auth_token_required
+    @roles_accepted(AccountRole.ISLAND_INTERFACE.name)
+    def put(self):
+        try:
+            masque = request.json
+            self._agent_binary_service.set_masque(masque)
+            return {}, HTTPStatus.NO_CONTENT
+
```

Comment:
Is this going to accept base64 ??

Reply:
We're going to stick to raw bytes or octets.